File: app/src/main/java/com/imfondof/wanandroid/ui/system/SystemFrg.java
package com.imfondof.wanandroid.ui.system;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;
import com.imfondof.wanandroid.R;
import com.imfondof.wanandroid.data.bean.SystemResult;
import com.imfondof.wanandroid.data.bean.WanHomeListBean;
import com.imfondof.wanandroid.data.http.HttpClient;
import com.imfondof.wanandroid.other.utils.ListUtil;
import com.imfondof.wanandroid.other.utils.SPUtils;
import com.imfondof.wanandroid.ui.base.MyFragmentPagerAdapter;
import com.imfondof.wanandroid.ui.base.BaseFragment;
import com.imfondof.wanandroid.ui.index.WanHomeFrg;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class SystemFrg extends BaseFragment {
    private TextView titleTv;
    private RecyclerView systemRv;
    private SystemAdapter systemAdapter;

    public static Fragment newInstance() {
        Bundle bundle = new Bundle();
        Fragment fragment = new SystemFrg();
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public int setContent() {
        return R.layout.frg_system;
    }

    @Override
    protected void initView() {
        super.initView();
        titleTv = getView(R.id.title_tv);
        titleTv.setText("体系");
        systemRv = getView(R.id.recycler_view);

        getSystem();

        systemAdapter = new SystemAdapter();
        systemRv.setLayoutManager(new LinearLayoutManager(getContext()));
        systemRv.setAdapter(systemAdapter);
    }

    public void getSystem() {
        Call<SystemResult> call = HttpClient.Builder.getWanAndroidService().getSystem();
        call.enqueue(new Callback<SystemResult>() {
            @Override
            public void onResponse(Call<SystemResult> call, Response<SystemResult> response) {
                systemAdapter.setNewData(response.body().data);
                dissmissLoding();
            }

            @Override
            public void onFailure(Call<SystemResult> call, Throwable t) {
                dissmissLoding();
            }
        });
    }
}

File: app/src/main/java/com/imfondof/wanandroid/ui/system/SystemFrg.kt
package com.imfondof.wanandroid.ui.system

import android.os.Bundle
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.imfondof.wanandroid.R
import com.imfondof.wanandroid.data.http.HttpClient
import com.imfondof.wanandroid.other.ext.id
import com.imfondof.wanandroid.ui.base.BaseFragment
import com.scwang.smartrefresh.layout.SmartRefreshLayout
import com.scwang.smartrefresh.layout.api.RefreshLayout
import com.scwang.smartrefresh.layout.listener.OnRefreshListener
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class SystemFrg : BaseFragment(), OnRefreshListener {
    private var mTreeAdapter: TreeAdapter? = null

    private val titleTv: TextView by id(R.id.title_tv)
    private val systemRv: RecyclerView by id(R.id.recycler_view)
    private val smartRefresh: SmartRefreshLayout by id(R.id.refresh_layout)

    override fun setContent() = R.layout.frg_system

    companion object {
        @JvmStatic
        fun newInstance(): Fragment {
            val bundle = Bundle()
            val fragment: Fragment = SystemFrg()
            fragment.arguments = bundle
            return fragment
        }
    }

    override fun initView() {
        super.initView()
        titleTv.text = "体系"

        smartRefresh.setEnableLoadMore(false)
        smartRefresh.setOnRefreshListener(this)
        mTreeAdapter = TreeAdapter(activity)
        systemRv.layoutManager = LinearLayoutManager(activity)
        systemRv.adapter = mTreeAdapter
        getSystem()
    }

//    override fun loadData() {
//        super.loadData()
//        getSystem()
//    }

    private fun getSystem() {
        val call = HttpClient.Builder.getWanAndroidService().system
        call.enqueue(object : Callback<TreeBean> {
            override fun onResponse(call: Call<TreeBean>, response: Response<TreeBean>) {
                mTreeAdapter!!.setNewData(response.body()?.data)
                dissmissLoding()
            }

            override fun onFailure(call: Call<TreeBean>, t: Throwable) {
                dissmissLoding()
            }
        })
    }

    override fun onRefresh(refreshLayout: RefreshLayout) {
        getSystem()
    }
}